app/models/concerns/system_like.rb

```diff
@@ -40,4 +40,16 @@ def compliance_score
     score = (100 * (rules_passed.to_f / (rules_passed + rules_failed)))
     score.nan? ? 0.0 : score
   end
+
+  def selected_rules(profile = nil, selected_columns = [])
+    rules_for_system = profile.present? ? profile.rules : rules
+
+    rules_for_system.select(selected_columns).yield_self do |temp_rules|
+      temp_rules.where(
+        id: RuleResult.selected.where(
+          host_id: id
+        ).pluck(:rule_id)
```

Comment:
Nit: I think this should be defined as a variable outside the `select` loop so we don't run it N times.

Reply:
That's a good point - IDK if the interpreter would fix that automatically but regardless I'd rather just make sure. I'll update the PR with that.